### PR TITLE
Document reproducibility (#298)

### PR DESCRIPTION
**Pull request by tomwhite**
Happy to hear any suggestions on wording, etc.

I also updated the 'Is there GPU or multicore-CPU support?' FAQ since it was related.

### CODE REVIEW

**Comment by sleighsoft** [comment]
Adding reference #298

File: doc/faq.rst [hunk 1]
```diff
@@ -95,26 +95,38 @@ issues.
...
+representation is performed via a (slightly custom) stochastic gradient
+descent. SGD in UMAP can take advantage of multicore, but only if
+`random_state` is set to `None` (as explained in the next question).
+
+Is the output of UMAP reproducible?
+-----------------------------------
+
+Yes. By default the random seed used by UMAP is set to a fixed value so that
+the resulting output embedding is the same each time if run repeatedly on
+the same input. Having reproducible output is very useful to identically
```
Comment (sleighsoft) [inline doc/faq.rst]: Maybe better rename to `reproducible visual output` to avoid confusion. See discussion on twitter linked by @lmcinnes

File: umap/umap_.py [hunk 2]
```diff
@@ -1253,11 +1253,13 @@ class UMAP(BaseEstimator):
...
 
-    random_state: int, RandomState instance or None, optional (default: None)
+    random_state: int, RandomState instance or None, optional (default: 42)
         If int, random_state is the seed used by the random number generator;
         If RandomState instance, random_state is the random number generator;
         If None, the random number generator is the RandomState instance used
-        by `np.random`.
+        by `np.random`. Furthermore, setting to None allows UMAP to make use
+        of multiple cores for some parts of the algorithm that means it runs
+        faster, but at the expense of reproducibility.
```
Comment (sleighsoft) [inline umap/umap_.py]: Again, maybe rephrase.

File: doc/faq.rst (hunk 1 above)
Comment (tomwhite) [reply doc/faq.rst]: Changed as suggested

File: umap/umap_.py (hunk 2 above)
Comment (tomwhite) [reply umap/umap_.py]: Not sure "visually" is needed here.

**Comment by sleighsoft** [comment]
@tomwhite So `random_state=None` seems the way to go in the discussion. Sorry for the trouble tom.
